src/Traits/DirectiveTrait.php

```diff
@@ -13,32 +13,32 @@
 
 namespace App\Traits;
 
-use App\Helper\Helper;
 use App\Rst\RstParser;
+use App\Value\Lines;
 
 trait DirectiveTrait
 {
-    private function in(string $directive, \ArrayIterator $lines, int $number, array $directiveTypes = null): bool
+    private function in(string $directive, Lines $lines, int $number, array $directiveTypes = null): bool
     {
-        $lines = Helper::cloneIterator($lines, $number);
```

Comment:
I did this because otherwise the injected iterator does not get back to its initial position, does this work now?

Reply:
Well, in the `AnalyseCommand` there is still a separate Iterator being used. The Rules are using the `seek`-method so they can jump back themselves.